### PR TITLE
feat: limit appliable constraints

### PR DESCRIPTION
**Pull request by brycetham**
This PR limits the constraints that can be passed to the `applyConstraints` method in both `LocalAudioStream` and `LocalVideoStream`.

### CODE REVIEW

File: src/media/local-audio-stream.ts [hunk 1]
```diff
@@ -1,20 +1,10 @@
 import { logger } from '../util/logger';
 import { LocalStream, LocalStreamEventNames } from './local-stream';
 
-// TODO: for applyConstraints, we only support things which can be changed on the
-// track itself.  This would not include 'deviceId', and maybe more of these values.
-// Might need to define one AudioConstraints for use with getUserMedia, and refine
-// it even further for applyConstraints here.
-export type AudioConstraints = Pick<
```
Comment (bbaldino) [inline src/media/local-audio-stream.ts:8]: Do we have these defined somewhere? The 'full set' should be usable in the getUserMedia calls

Comment (brycetham) [reply src/media/local-audio-stream.ts:8]: We do have `AudioDeviceConstraints` and `VideoDeviceConstraints` defined in `device-management.ts`, if that's what you're asking.

Comment (brycetham) [reply src/media/local-audio-stream.ts:8]: Unrelated, but just noticed that we're not picking from `MediaTrackConstraints` when defining `VideoDeviceConstraints` like we do with `AudioDeviceConstraints`. I can make that change real fast.

Comment (bbaldino) [reply src/media/local-audio-stream.ts:8]: ok cool

**Review by bbaldino — APPROVED** [review]

**Pull request merged**